### PR TITLE
Hil refinements

### PR DESCRIPTION
**Pull request by bkueng**
- avoid some warnings in HIL mode, increase the sensor timeout
- remove unneeded code in state_machine_helper
- add a HIL config for RPI (did not fly stable because I used it over WiFi)

edit: HIL works fine on the RPi when using a LAN cable

### CODE REVIEW

File: src/modules/sensors/voted_sensors_update.cpp [hunk 1]
```diff
@@ -82,6 +82,11 @@ VotedSensorsUpdate::VotedSensorsUpdate(const Parameters &parameters, bool hil_en
 	_baro.voter.set_timeout(300000);
 	_mag.voter.set_timeout(300000);
 	_mag.voter.set_equal_value_threshold(1000);
+
+	if (_hil_enabled) { // HIL has less accurate timing so increase the timeouts a bit
+		_gyro.voter.set_timeout(200000);
+		_accel.voter.set_timeout(200000);
```
Comment (julianoes) [inline src/modules/sensors/voted_sensors_update.cpp]: Where is the initial timeout set? Shouldn't it be there?

File: src/modules/sensors/sensors.cpp [hunk 2]
```diff
@@ -332,8 +332,10 @@ Sensors::parameters_update()
 
 	// TODO: this needs fixing for QURT and Raspberry Pi
 	if (!h_baro.isValid()) {
-		PX4_ERR("no barometer found on %s (%d)", BARO0_DEVICE_PATH, h_baro.getError());
-		ret = PX4_ERROR;
+		if (!_hil_enabled) { // in HIL we don't have a baro
+			PX4_ERR("no barometer found on %s (%d)", BARO0_DEVICE_PATH, h_baro.getError());
```
Comment (julianoes) [inline src/modules/sensors/sensors.cpp]: Shouldn't we have a fake baro?

File: src/modules/commander/state_machine_helper.cpp [hunk 3]
```diff
@@ -493,138 +493,8 @@ transition_result_t hil_state_transition(hil_state_t new_state, orb_advert_t sta
 			    || current_status->arming_state == vehicle_status_s::ARMING_STATE_STANDBY
 			    || current_status->arming_state == vehicle_status_s::ARMING_STATE_STANDBY_ERROR) {
 
-#ifdef __PX4_NUTTX
```
Comment (julianoes) [inline src/modules/commander/state_machine_helper.cpp]: Interesting stuff, so we don't want to use this blocking/unblocking in the future?

File: posix-configs/rpi/px4_hil.config [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# HITL configuration
+# connect to it with jMAVSim:
+# ./Tools/jmavsim_run.sh -q -i <IP> -p 14577 -r 250
```
Comment (julianoes) [inline posix-configs/rpi/px4_hil.config]: :+1:

File: src/modules/sensors/voted_sensors_update.cpp (hunk 1 above)
Comment (bkueng) [inline src/modules/sensors/voted_sensors_update.cpp]: It's just the default from set in the voters (I think 20ms)

File: src/modules/sensors/sensors.cpp (hunk 2 above)
Comment (bkueng) [inline src/modules/sensors/sensors.cpp]: We have a fake baro but it is directly published via mavlink, there is no `BARO0_DEVICE_PATH` for it

File: src/modules/commander/state_machine_helper.cpp (hunk 3 above)
Comment (bkueng) [inline src/modules/commander/state_machine_helper.cpp]: I don't think it's needed, at least not here. blocking is still interesting for example if you want to test sensor failover.

File: src/modules/sensors/voted_sensors_update.cpp (hunk 1 above)
Comment (julianoes) [inline src/modules/sensors/voted_sensors_update.cpp]: Ok.

File: src/modules/sensors/sensors.cpp (hunk 2 above)
Comment (julianoes) [inline src/modules/sensors/sensors.cpp]: Got it.

File: src/modules/commander/state_machine_helper.cpp (hunk 3 above)
Comment (julianoes) [inline src/modules/commander/state_machine_helper.cpp]: Ah right, ok.

**Comment by bkueng** [comment]
I'll do that. gps is not started in HIL mode: https://github.com/PX4/Firmware/blob/master/ROMFS/px4fmu_common/init.d/rcS#L402

**Comment by bkueng** [comment]
Tested, AirSim still works

**Pull request merged**